client/state/billing-data/selectors.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { get, find, mapValues } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+import { parseTransactionDate } from './util';
+
+/**
+ * Returns true if we are currently making a request to get the billing data.
+ * False otherwise.
+ *
+ * @param  {Object}    state  Global state tree
+ * @return {Boolean}          Whether the billing data is being requested
+ */
+export function isRequestingBillingData( state ) {
+	return get( state, 'billingData.requesting', false );
+}
+
+/**
+ * Returns all billing data.
+ * Returns null if the billing data has not been fetched yet.
+ *
+ * @param  {Object}  state   Global state tree
+ * @return {?Object}         Billing data
+ */
+export const getBillingData = createSelector(
+	( state ) => {
+		const allTransactions = get( state, 'billingData.items', null );
+		if ( ! allTransactions ) {
+			return null;
+		}
+
+		return mapValues( allTransactions, transactions => transactions.map( parseTransactionDate ) );
+	}
```

Comment:
The [default](https://github.com/Automattic/wp-calypso/blob/master/client/lib/create-selector/index.js#L28) will clear out data when any state changes